wayland.c

```diff
@@ -381,24 +387,25 @@ bool init_wayland(struct mako_state *state) {
 		}
 		wl_display_roundtrip(state->display);
 	}
-	if (state->xdg_output_manager == NULL &&
-			strcmp(state->config.output, "") != 0) {
-		fprintf(stderr, "warning: configured an output but compositor doesn't "
-			"support xdg-output-unstable-v1 version 2\n");
+	if (state->xdg_output_manager == NULL) {
+		struct mako_surface *surface;
+		wl_list_for_each(surface, &state->surfaces, link) {
+			if (strcmp(surface->config->output, "") != 0) {
+				fprintf(stderr, "warning: configured an output "
+					"but compositor doesn't support "
+					"xdg-output-unstable-v1 version 2\n");
```

Comment:
I'd either break after logging this once or add the surface name to the message.